gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspConfigurations.kt

```diff
@@ -66,13 +74,17 @@ class KspConfigurations(private val project: Project) {
             is KotlinMultiplatformExtension -> {
                 kotlin.targets.configureEach(::decorateKotlinTarget)
 
-                // Adding multiplatform configuration removed support for the root ksp configuration.
-                // Try to make this breaking change less breaking by adding a clear error.
-                project.configurations.create("ksp").dependencies.all {
-                    throw InvalidUserCodeException(
-                        "The 'ksp' configuration cannot be used in Kotlin Multiplatform projects. " +
+                project.afterEvaluate {
```

Comment:
Avoid usage of `afterEvaluate` as the lifecycle is tricky, to say the least. Instead, you can have `configuartion.dependencies.whenObjectAdded { reportIssueOnce(...) }`.